Fix questa hjson file with needed options

mostafaabdallah12 opened the pull request:
We've been working for the last two months to get OpenTitan to work with Questa, and we have succeeded! 

We changed some setup files to add questa options and setups.

This is one of the setup files: hw/dv/tools/dvsim/questa.hjson

Once it's accepted in OpenTitan master branch we will push the remaining setup files.

Commit:
Fix questa hjson file with needed options

Signed-off-by: Mostafa  Hassan <mostafha@orw-mostafha-vm.wv.mentorg.com>

rswarbrick commented on hw/dv/tools/dvsim/questa.hjson:
Nit: I think the indentation is off here?

rswarbrick commented on hw/dv/tools/dvsim/questa.hjson:
I'm willing to believe this is needed, but could it have a comment above it that says roughly we're suppressing? Before this change, `run_opts` had a comment like that for the 8323 message that was being suppressed.

I think we will need equivalent notes for all the other warnings that are being suppressed.

Commit:
Add Questa setup needed changes, Add notes for each suppressed error

Commit:
Add Questa setup needed changes, Add notes for each suppressed error, fix identation

mostafaabdallah12 commented:
@rswarbrick  I fixed the indentation issue. and added notes for the suppressed errors.

rswarbrick commented on hw/dv/tools/dvsim/questa.hjson:
I'm really sorry, but would you mind adding notes above this line too?

rswarbrick commented on hw/dv/tools/dvsim/questa.hjson:
Eek! The waiver looks perfectly sensible, but is it easy for you to generate a list of examples that currently fail? I don't have a Questa licence, but I'd be very happy to tidy this up...

rswarbrick replied:
I'm working (very slowly) through this list of waivers. I think the first one will be something we can remove once #26002 is merged. (If not, thank you for testing! Please tell me what I've missed!)

rswarbrick commented on hw/dv/tools/dvsim/questa.hjson:
This is very surprising to me! That would presumably be a bug in the code in question? Can you send examples that are currently failing this?

rswarbrick replied:
Again, please could you provide a list of examples where this doesn't work? (Or alternatively a pattern that will find them with grep?)

rswarbrick commented on hw/dv/tools/dvsim/questa.hjson:
Again, this looks rather surprising to me! I assume that `bit [7:0] foo = "12" won't set `foo` to `8'hc` :-) Can you send a list of examples that currently fail this?

rswarbrick reviewed (commented):
I started adding notes to this but realised that I was writing the same thing many times over :-)

Would you mind giving a list of examples for each waived rule? We might just be able to fix them, or otherwise should probably file issues to track fixing them properly.

mostafaabdallah12 commented:
@rswarbrick  Can we proceed with the pull request and add the change and I will file issues for all suppressible errors we added for with more details

rswarbrick commented:
Sorry for not replying more quickly. I'd really appreciate at least a list of what the magic IDs mean. I don't have a Questa licence, so I don't know what the numbers mean at all. The output could be an issue (generated by me) to track the fact that this needs sorting out at our end.

Commit:
Fix questa hjson file with needed options, added notes for all suppressed errors

mostafaabdallah12 commented:
@rswarbrick  I thought we had added notes for all IDs. I updated the last commit to have notes for all the IDs. 
It's worth mentioning that it's not a must that it's an issue in code. Maybe Questa couldn't retrieve this info and throws an error, which makes sense to have these IDs as suppressible IDs. I will review from our side and make sure we file issues on github in case there's a real issue not questa issue.

rswarbrick commented on hw/dv/tools/dvsim/questa.hjson:
Oof, that sounds like a genuine bug in some DV code ("passing a null pointer to something that wants an integer"). Can you give a list of examples of this? I'd love to just fix them!

rswarbrick reviewed (commented):
As a very general review, rather than adding waivers I'd *much* prefer to just fix the code.

If you can list examples of where we generate any of the errors, I'd be very happy to sort them out.
